com.unity.multiplayer.mlapi/Runtime/Core/NetworkUpdateManager.cs

```diff
@@ -4,216 +4,193 @@
-            }
-        }
-
-        public void FixedUpdateRegister(Action updateAction)
-        {
-            FixedUpdateAction = updateAction;
-        }
+                Type InsertAfterType = typeof(EarlyUpdate.UnityWebRequestUpdate);
+                switch(updateStage.Key)
+                {
+                    case NetworkUpdateStages.PREUPDATE:
+                        {
+                            InsertAfterType = typeof(EarlyUpdate.UnityWebRequestUpdate);
+                            break;
+                        }
+                    case NetworkUpdateStages.FIXEDUPDATE:
+                        {
+                            InsertAfterType = typeof(FixedUpdate.AudioFixedUpdate);
+                            break;
+                        }
+                    case NetworkUpdateStages.UPDATE:
+                        {
+                            InsertAfterType = typeof(PreUpdate.UpdateVideo);
+                            break;
+                        }
+                    case NetworkUpdateStages.LATEUPDATE:
+                        {
+                            InsertAfterType = typeof(PostLateUpdate.ProcessWebSendMessages);
+                            break;
+                        }
+                }
 
-        public void Update()
-        {
-            if (UpdateAction != null)
-            {
-                UpdateAction.Invoke();
+                 InsertSystem(ref def, updateStage.Value, InsertAfterType);
             }
-        }
 
-        public void UpdateRegister(Action updateAction)
-        {
-            UpdateAction = updateAction;
+#if UNITY_EDITOR
+            PrintPlayerLoop(def);
+#endif
+
+            PlayerLoop.SetPlayerLoop(def);
         }
 
-        public void PostUpdate()
+        /// <summary>
+        /// OnNetworkLoopSystemDestroyed
+        /// This should be invoked by the registered INetworkUpdateLoopSystem when the class is being destroyed or it no longer wants to receive updates
+        /// </summary>
+        /// <param name="networkLoopSystem"></param>
+        public static void OnNetworkLoopSystemDestroyed(INetworkUpdateLoopSystem networkLoopSystem)
         {
-            if (PostUpdateAction != null)
+            if (s_RegisteredUpdateLoopSystems.ContainsKey(networkLoopSystem))
             {
-                PostUpdateAction.Invoke();
+                var def = PlayerLoop.GetCurrentPlayerLoop();
+                foreach (KeyValuePair<NetworkUpdateStages,PlayerLoopSystem> updateStage in s_RegisteredUpdateLoopSystems[networkLoopSystem])
+                {
+                    if(!RemoveSystem(ref def,updateStage.Value))
+                    {
+                        Debug.LogWarning(updateStage.Value.type.Name + " tried to remove itself but no instsance was found!!!");
+                    }
+                }
+
+                s_RegisteredUpdateLoopSystems.Remove(networkLoopSystem);
+#if UNITY_EDITOR
+                PrintPlayerLoop(def);
+#endif
```

Comment:
do we really want to _always_ print player loop all the time when running in the editor?

Reply:
Well, I was thinking it is useful to get registered player state updates in the editor whenever one changes.  This should most likely be exposed as a bool property that can be enabled.  However,  this might need to be an alternate discussion on "general MLAPI settings" that don't fall under the NetworkingManager context but have no real editor/object representation.   So, a new task (MTT-422) has been made for this and several other areas that could stand to be better accessible to users.